services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/client/pulsar/TopicManager.java

```diff
@@ -15,9 +15,72 @@
 */
 package com.futurewei.alcor.dataplane.client.pulsar;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Configuration;
+
+import javax.annotation.PostConstruct;
+import java.util.*;
+
+@Configuration
+@ConditionalOnProperty(prefix = "mq", name = "type", havingValue = "pulsar")
 public class TopicManager {
+    private static final Logger LOG = LoggerFactory.getLogger(TopicManager.class);
+
+    @Autowired
+    private PulsarConfiguration configuration;
+
+    private Map<String, String> hostIpToGroupTopic;
+
+    private Map<String, String> groupTopicToMulticastTopic;
+
+    private Set<String> groupTopics;
+
+    private Set<String> multicastTopics;
+
+    @PostConstruct
+    public void init() throws Exception {
+        try {
+            this.hostIpToGroupTopic = parseTopicConfig(configuration.getHostIpToGroupTopicMap());
+            this.groupTopicToMulticastTopic = parseTopicConfig(configuration.getGroupTopicToMulticastTopicMap());
+        }catch (Exception e) {
+            throw new Exception("Parse topic config error: " + e);
```

Comment:
We can define an explicit exception for this parsing error. Please find an example in https://github.com/futurewei-cloud/alcor/tree/master/services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/exception 

The idea is that we know every possible exception our own program may throw, and define it before hand. In each of the predefined exception, we can define its error code and message, and come to the debugging time, we can quickly locate where the exception is thrown and why based on the given error code and message.

Reply:
I will add a predefined exception to solve this problem.